backend/src/shortest-path/shortest-path.service.ts

```diff
@@ -0,0 +1,380 @@
+        else {
+          const vEdgeId = getNextVEdgeId();
+          // Do not take the real id of the last node, but the allocated virtual id.
+          let from = lastNode.id;
+          let to = currNodeId;
+
+          // The current edge points from curr to last (the reverse of the path direction)
+          if (currNodeId === currEdge.from) {
+            const acc = from;
+            from = to;
+            to = acc;
+          }
+
+          queryResult.edges.push({
+            id: vEdgeId,
+            from,
+            to,
+            type: '',
+            virtual: true,
+            isPath: true,
+          });
+        }
+
+        lastNode = currNode;
+      }
+      // The target node was not-found. Check whether the source node is virtual. If it is
+      // we don't have to do anything, as the virtual node already replaces a complete
+      // subgraph (that our node now will become part of), if it is not, we have to create
+      // it as a replacement for the subgraph. In the last case, we also have to insert
+      // a virtual edge.
+      else if (!lastNode.virtual) {
+        const vEdgeId = getNextVEdgeId();
+        const vNodeId = getNextVNodeId();
+        let from = lastNodeId;
+        let to = vNodeId;
+
+        // The current edge points from curr to last (the reverse of the path direction)
+        if (currNodeId === currEdge.from) {
+          const acc = from;
+          from = to;
+          to = acc;
+        }
+
+        queryResult.edges.push({
+          id: vEdgeId,
+          from,
+          to,
+          type: '',
+          virtual: true,
+          isPath: true,
+        });
+
+        lastNode = {
+          id: vNodeId,
+          types: [],
+          virtual: true,
+          isPath: true,
+        };
+
+        queryResult.nodes.push(lastNode);
+      }
+
+      lastNodeId = currNodeId;
+    }
+
+    return queryResult;
+  }
+
+  public async findShortestPath(
+    startNode: number,
+    endNode: number,
+    ignoreEdgeDirections?: boolean
+  ): Promise<Path | null> {
+    if (startNode === endNode) {
+      const nodeDetails = (
+        await this.queryService.getNodesById([startNode])
+      ).find(() => true);
+
+      if (!nodeDetails) {
+        return null;
+      }
+
+      return {
+        start: { id: nodeDetails.id, types: nodeDetails.types },
+        end: { id: nodeDetails.id, types: nodeDetails.types },
+        edges: [],
+      };
+    }
+
+    const graphName = ignoreEdgeDirections
+      ? KMAP_GDS_GRAPH_NAME_SHORTEST_PATH
+      : KMAP_GDS_GRAPH_NAME_SHORTEST_PATH_DIRECTED;
+
+    if (!(await this.existsGDSProjection(graphName))) {
+      await this.createGDSProjection(graphName, ignoreEdgeDirections);
+    }
+
+    // eslint-disable-next-line no-return-await
+    return await this.execGDSShortestPath(
+      graphName,
+      startNode,
+      endNode,
+      ignoreEdgeDirections
+    );
+  }
+
+  private async createGDSProjection(
```

Comment:
You should definitely explain what these GDSProjections are. In my opinion you don't need to doc every method here with tsdoc since they are private helpermethods, but currently, someone who has never heard of GDSProjections does not know what this is used for